### PR TITLE
add github actions on merge and PR requests

### PR DESCRIPTION
**Pull request by SoloSynth1**

### CODE REVIEW

**Comment by SoloSynth1** [comment]
Build failed as `Dockerfile` is not present in the `main` branch. Possibly fixed by #39.

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Publish Docker image
```
Comment (phchen5) [inline .github/workflows/build-and-test.yaml:1]: Change to a more suitable name for this task

**Review by phchen5 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-test.yaml (hunk 1 above)
Comment (SoloSynth1) [reply .github/workflows/build-and-test.yaml]: @phchen5 changed, please review again

**Review by phchen5 — APPROVED** [review]
lgtm

**Pull request merged**